hana/lib/drivers/hdb.js

```diff
@@ -9,7 +9,7 @@ const iconv = require('iconv-lite')
 const { driver, prom, handleLevel } = require('./base')
 const { isDynatraceEnabled: dt_sdk_is_present, dynatraceClient: wrap_client } = require('./dynatrace')
 
-if (cds.env.features.sql_simple_queries === 3) {
+if (cds.env.features.sql_simple_queries) {
```

Comment:
**OK, let's see:**

`features.sql_simple_queries: 1`: Not necessary, as always coming as boolean, right? And is default. Here we don't want to patch hdb
`features.sql_simple_queries: 2`: Does not make sense anymore at all, as not needed for HANA (sq3) and accepted for sqlite (sq1)
`features.sql_simple_queries: 3`: Yes, here it should be applied

Reply:
> This PR achieves the same behavior for all databases without having to differ between sql_simple_queries:1|2|3 by having all databases return true and false for Booleans.

Yes, that's good ! But I would not extra patch `hdb`, when anyways the booleans come correctly from query ...

Reply:
They don't come correctly from the query when using `hdb` as the protocol version it uses give the metadata as tinyint not as Boolean. That is why it is being patched instead of rendering `json` queries.

I tried to find the place where `hdb` defines the protocol version, but was not successful to change the behavior of HANA to match that for hana-client. A protocol upgrade probably comes with more features then just native Boolean support. So a feature request to `hdb` will most definitely be rejected.

Reply:
I thought that boolean comes from JSON function for `simple_queries: 1`, isn't it?